File: app/src/main/java/me/imzack/app/end/view/contract/TypeSearchViewContract.java
package me.imzack.app.end.view.contract;

import me.imzack.app.end.view.adapter.TypeSearchListAdapter;

public interface TypeSearchViewContract extends BaseViewContract {

    void showInitialView(int typeCount, TypeSearchListAdapter typeSearchListAdapter);

    void onSearchChanged(boolean isNoSearchInput, boolean isTypeSearchEmpty);

    void onTypeItemClicked(int typeListPos);
}

File: app/src/main/kotlin/me/imzack/app/end/view/contract/TypeSearchViewContract.kt
package me.imzack.app.end.view.contract

import me.imzack.app.end.view.adapter.TypeSearchListAdapter

interface TypeSearchViewContract : BaseViewContract {

    fun showInitialView(typeCount: Int, typeSearchListAdapter: TypeSearchListAdapter)

    fun onSearchChanged(isNoSearchInput: Boolean, isTypeSearchEmpty: Boolean)

    fun onTypeItemClicked(typeListPos: Int)
}
